File: engine/src/com/magma/engine/ui/dialog/DialogMessage.java
package com.magma.engine.ui.dialog;

public class DialogMessage {
	
	public static final String PLACEHOLDER = "No dialog found! Whoops...";
	
	public final DialogStyle style;
	public final String message;
	
	public DialogMessage(int key, DialogStyle style) {
		this.style = style;
		this.message = PLACEHOLDER;
	}
	
	public DialogMessage(int key) {
		this.message = PLACEHOLDER;
		this.style = DialogStyle.Basic;
	}

	public DialogMessage(String message, DialogStyle style) {
		this.message = message;
		this.style = style;
	}
	
	public DialogMessage(String message) {
		this.message = message;
		this.style = DialogStyle.Basic;
	}
}

File: engine/src/com/magma/engine/ui/dialog/DialogMessage.kt
package com.magma.engine.ui.dialog

class DialogMessage {
    val style: DialogStyle
    val message: String

    constructor(key: Int, style: DialogStyle) {
        this.style = style
        message = PLACEHOLDER
    }

    constructor(key: Int) {
        message = PLACEHOLDER
        style = DialogStyle.Basic
    }

    constructor(message: String, style: DialogStyle) {
        this.message = message
        this.style = style
    }

    constructor(message: String) {
        this.message = message
        style = DialogStyle.Basic
    }

    companion object {
        const val PLACEHOLDER = "No dialog found! Whoops..."
    }
}